### PR TITLE
Add Soboleva Modified Hyberbolic Tangent function

### PR DESCRIPTION
**Pull request by saahil-mahato**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by haxkd — APPROVED** [review]

File: neural_network/activation_functions/sobovela_modified_hyperbolic_tangent.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+def sobovela_modified_hyperbolic_tangent(
+    vector: np.ndarray, a_value: float, b_value: float, c_value: float, d_value: float
+) -> np.ndarray:
+    """
+    Implements the Sobovela Modified Hyperbolic Tangent function
+
+    Parameters:
+        vector (ndarray): A vector that consitis of numeric values
+        alpha (float): alpha value
+        beta (float): beta value
```
Comment (tianyizheng02) [inline neural_network/activation_functions/sobovela_modified_hyperbolic_tangent.py:24]: This section doesn't match the actual parameters of the function

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
The function is called **Soboleva**, not Sobovela

File: neural_network/activation_functions/sobovela_modified_hyperbolic_tangent.py (hunk 1 above)
Comment (saahil-mahato) [reply neural_network/activation_functions/sobovela_modified_hyperbolic_tangent.py]: Thanks

**Comment by saahil-mahato** [comment]
> The function is called **Soboleva**, not Sobovela

Thanks. I have edited filename and docs

File: neural_network/activation_functions/soboleva_modified_hyperbolic_tangent.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def soboleva_modified_hyperbolic_tangent(
+    vector: np.ndarray, a_value: float, b_value: float, c_value: float, d_value: float
+) -> np.ndarray:
+    """
+    Implements the Soboleva Modified Hyperbolic Tangent function
+
+    Parameters:
+        vector (ndarray): A vector that consitis of numeric values
```
Comment (tianyizheng02) [inline neural_network/activation_functions/soboleva_modified_hyperbolic_tangent.py:22]:
```suggestion
        vector (ndarray): A vector that consists of numeric values
```
Typo

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**